src/flint/flint_base/flint_base.pyx

```diff
@@ -869,6 +941,87 @@ cdef class flint_mpoly(flint_elem):
         """
         return zip(self.monoms(), self.coeffs())
 
+    def drop_unused_gens(self):
+        """
+        Remove unused generators from this polynomial. Returns a potentially new
+        context, and potentially new polynomial.
+
+        A generator is unused if it's maximum degree is 0.
+
+            >>> from flint import fmpz_mpoly_ctx
+            >>> ctx = fmpz_mpoly_ctx.get(('x', 4))
+            >>> ctx2 = fmpz_mpoly_ctx.get(('x1', 'x2'))
+            >>> f = sum(ctx.gens()[1:3])
+            >>> f
+            x1 + x2
+            >>> new_ctx, new_f = f.drop_unused_gens()
+            >>> new_ctx
+            fmpz_mpoly_ctx(2, '<Ordering.lex: 'lex'>', ('x1', 'x2'))
+            >>> new_f
+            x1 + x2
+        """
+        new_ctx = self.context().drop_gens(
+            *(i for i, x in enumerate(self.degrees()) if not x)
+        )
+        return new_ctx, self.coerce_to_context(new_ctx)
+
+    def coerce_to_context(self, other_ctx, mapping: dict[str | int, str | int] = None):
+        """
+        Coerce this polynomial to a different context.
+
+        This is equivalent to composing this polynomial with the generators of another
+        context. By default the mapping between contexts is inferred based on the name
+        of the generators. Generators with names that are not found within the other
+        context are mapped to 0. The mapping can be explicitly provided.
```

Comment:
Mapping the unused generators to zero makes this not really what I would call a "coercion". This is perhaps better described as a "projection".

For a coercion I would expect an error if the element cannot be represented in the target context.

Reply:
That's fair enough. I've renamed it to `project_to_context`. 

> For a coercion I would expect an error if the element cannot be represented in the target context.

It could certainly check if that was the case and raise a warning/exception